packages/sync/src/modules/class-users.php

```diff
@@ -233,11 +233,14 @@ public function expand_user( $user ) {
 	 */
 	public function get_real_user_capabilities( $user ) {
 		$user_capabilities = array();
-		if ( is_wp_error( $user ) ) {
+		if ( is_wp_error( $user ) || ! is_a( $user, 'WP_User' ) ) {
 			return $user_capabilities;
 		}
+
+		$user_caps = $user->allcaps;
+
 		foreach ( Defaults::get_capabilities_whitelist() as $capability ) {
-			if ( user_can( $user, $capability ) ) {
+			if ( in_array( $capability, $user_caps, true ) ) {
 				$user_capabilities[ $capability ] = true;
 			}
 		}
```

Comment:
Would it make sense to do something like this instead?
```
$caps = array_intersect( Defaults::get_capabilities_whitelist(), $user->allcaps );
$user_capabilities = array_fill_keys( $caps, true );
```
That should give the same result assuming the capabilities are all supposed to be strings anyway.

Reply:
This made me realize that I wasn't treating the `$user_caps` array properly. I ended up committing e5fc8a2904e0af5c8311bbcbf60753cc969508d9.